src/nxt_cgroup.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) Andrew Clayton
+ * Copyright (C) F5, Inc.
+ */
+
+#ifndef _NXT_CGROUP_H_INCLUDED_
+#define _NXT_CGROUP_H_INCLUDED_
+
+
+#include <nxt_main.h>
+#include <nxt_process.h>
+
+
+typedef struct nxt_cgroup_s  nxt_cgroup_t;
+/*
+ * We need nxt_process.h for the definition of nxt_process_t,
+ * however we need to include this header file in nxt_process.h
+ * for the definition of nxt_cgroup_t and due to the header guards
+ * nxt_process.h won't be processed at this point so we don't get
+ * the definition of nxt_process_t.
+ *
+ * This forward delcaration of nxt_process_t allows it to compile.
+ */
+typedef struct nxt_process_s nxt_process_t;
```

Comment:
Could you please try to fix this circular dependency by using a third header breaking either nxt_process.h or nxt_cgroup.h?  I'd like to avoid repeated declarations.

Reply:
I'll give it some thought.

Reply:
For now I've put the definition (and typedef) of nxt_cgroup_s into src/nxt_process.h which is where that structure is embedded in nxt_process_isolation_t. Kind of like nxt_process_automount_t.

Reply:
Makes sense.  I'm not entirely happy that we are forced to workaround it like that, but the solution would be to nuke the headers, and that's not happening soon. ;)

Reply:
It seemed like currently the best alternative...